File: java/java-starter/src/main/java/com/showmeyourcode/kafka/java/starter/Starter.java
package com.showmeyourcode.kafka.java.starter;

import com.showmeyourcode.kafka.java.kafka_consumer.JavaKafkaConsumer;
import com.showmeyourcode.kafka.java.kafka_producer.JavaKafkaProducer;
import com.showmeyourcode.kafka.java.kafka_producer.JavaKafkaAvroProducer;

public class Starter {
    public static void main(String... args) {
        int numberOfMessages;
        if (args.length == 0) {
            numberOfMessages = 5;
        } else {
            numberOfMessages = Integer.parseInt(args[0]);
        }
        new JavaKafkaProducer.JavaKafkaProducerBuilder().withNumberOfMessage(numberOfMessages).build().produce();
        new JavaKafkaAvroProducer.JavaKafkaAvroProducerBuilder().withNumberOfMessage(numberOfMessages).withAvroClass().withName("Avro Class").build().produce();
        new JavaKafkaAvroProducer.JavaKafkaAvroProducerBuilder().withNumberOfMessage(numberOfMessages).withAvroClassFromFile().withName("Avro File").build().produce();

        new JavaKafkaConsumer.JavaKafkaConsumerBuilder().withNumberOfMessages(Long.MAX_VALUE).build().consume();
    }
}

File: kotlin/kotlin-starter/src/main/kotlin/com/showmeyourcode/kafka/kotlin/starter/Starter.kt
package com.showmeyourcode.kafka.kotlin.starter

import com.showmeyourcode.kafka.kotlin.consumer.KotlinKafkaConsumer
import com.showmeyourcode.kafka.kotlin.producer.KotlinKafkaAvroWithRegistryProducer
import com.showmeyourcode.kafka.kotlin.producer.KotlinKafkaProducer

fun main(args: Array<String>) {
    val numberOfMessages =
        if (args.isEmpty()) {
            10L
        } else {
            args[0].toLong()
        }

    KotlinKafkaProducer.KotlinKafkaProducerBuilder().withNumberOfMessages(numberOfMessages).build().produce()
    KotlinKafkaAvroWithRegistryProducer.KotlinKafkaAvroWithRegistryProducerBuilder().withNumberOfMessages(
        numberOfMessages,
    ).build().produce()

    KotlinKafkaConsumer.KotlinKafkaConsumerBuilder().withNumberOfMessages(numberOfMessages).build().consume()
}
